src/main/java/seedu/address/logic/parser/ArgumentMultimap.java

```diff
@@ -75,4 +75,14 @@ public void verifyNoDuplicatePrefixesFor(Prefix... prefixes) throws ParseExcepti
             throw new ParseException(Messages.getErrorMessageForDuplicatePrefixes(duplicatedPrefixes));
         }
     }
+
+    /**
+     * Returns true if all of the specified prefixes are present and contain non-empty values.
+     *
+     * @param prefixes The prefixes to check.
+     * @return true if all prefixes are present, false otherwise.
+     */
+    public boolean arePrefixesPresent(Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> this.getValue(prefix).isPresent());
+    }
```

Comment:
Will all the other `add/delete` related commands use this? @TechRaj

Reply:
I basically shifted what @hweinian  did for allergy here so more CRUD functions can use it and comply with DRY

Reply:
I think this shift is ok. It aligns with DRY and reduces code duplication. How about the rest of the parts that are using it? Did you fix those commands?